build.gradle.kts

```diff
@@ -0,0 +1,227 @@
+    showSimpleNames = false
+    showPassed = true
+    showSkipped = true
+    showFailed = true
+    showOnlySlow = false
+    showStandardStreams = false
+    showPassedStandardStreams = true
+    showSkippedStandardStreams = true
+    showFailedStandardStreams = true
+    logLevel = LogLevel.LIFECYCLE
+}
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// Dependencies
+
+dependencies {
+    api(platform(libs.spring.session.bom))
+    api(libs.spring.session.core)
+
+    api(platform(libs.spring.data.bom))
+    api(libs.spring.data.mongodb)
+
+    api(platform(libs.jackson.bom))
+    api(libs.jackson.databind)
+
+    api(platform(libs.spring.security.bom))
+    api(libs.spring.security.core)
+
+    // Optional dependencies
+    "optionalApi"(platform(libs.project.reactor.bom))
+    "optionalApi"(libs.project.reactor.core)
+
+    implementation(platform(libs.mongodb.driver.bom))
+    implementation(libs.mongodb.driver.core)
+    "optionalImplementation"(libs.mongodb.driver.sync)
+    "optionalImplementation"(libs.mongodb.driver.reactive.streams)
+
+    // We need the `libs.findbugs.jsr` dependency to stop `javadoc` from emitting
+    // `warning: unknown enum constant When.MAYBE`
+    //   `reason: class file for javax.annotation.meta.When not found`.
+    compileOnly(libs.findbugs.jsr)
+    errorprone(libs.nullaway)
+    errorprone(libs.google.errorprone.core)
+
+    testImplementation(platform(libs.junit.bom))
+    testImplementation(platform(libs.mockito.bom))
+    testImplementation(platform(libs.spring.framework.bom))
+
+    testImplementation(libs.bundles.testing)
+    testImplementation(libs.bundles.jakarta)
+    testImplementation(libs.bundles.spring.test)
+    testImplementation(libs.logback.core)
+    testImplementation(libs.project.reactor.test)
+
+    testRuntimeOnly(libs.junit.platform.launcher)
+}
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// Static Analysis
+
+tasks.named("check") { dependsOn("spotlessApply") }
+
+spotless {
+    java {
+        importOrder()
+
+        removeUnusedImports()
+
+        palantirJavaFormat(libs.versions.plugin.palantir.get()).formatJavadoc(true)
+
+        formatAnnotations()
+
+        // need to add license header manually to package-info.java and module-info.java
+        // due to the bug: https://github.com/diffplug/spotless/issues/532
+        licenseHeaderFile(file("config/spotless.license.java"))
+
+        targetExclude("build/generated/sources/buildConfig/**/*.java")
+    }
+
+    kotlinGradle {
+        ktfmt(libs.versions.plugin.ktfmt.get()).configure {
+            it.setMaxWidth(120)
+            it.setBlockIndent(4)
+        }
+        trimTrailingWhitespace()
+        leadingTabsToSpaces()
+        endWithNewline()
+    }
+
+    format("extraneous") {
+        target("*.xml", "*.yml", "*.md", "*.toml")
+        trimTrailingWhitespace()
+        leadingTabsToSpaces()
+        endWithNewline()
+    }
+}
+
```

Comment:
I assume the publishing block is done in another ticket?

Reply:
Yes